### ISSUE TITLE
active_record_pg multiverse suite fails when ran with parallelism

### ISSUE THREAD

**Issue by fallwith** [newrelic/newrelic-ruby-agent#1006]
Our GitHub Actions based CI system explicitly disables parallelism via the `SERIALIZE` environment variable, but elsewhere all of our unit and integration tests are expected to be able to operate in parallel.

At least with Ruby 3.0 and 3.1 and potentially with other Rubies, the `active_record_pg` multiverse suite will frequently fail when ran with parallelism (the default). For me, at least 1 out of every 3 runs will fail. Setting `SERIALIZE=1` eliminates the problem.

We should get that suite working with parallelism.

**Comment by fallwith** [comment]
Findings:

* There is a single failing Minitest test is involved, which is `test_metrics_for_calculation_methods` defined in `test/multiverse/suites/active_record_pg/active_record_test.rb`
* The problem will only occur when Rails versions < 7 are tested, given that `test_metrics_for_calculation_methods` only layers in a count with the assertion if ActiveRecord is below version 7.0
* The failing test is only attempted against Postgres, but a very similar test of the same name defined by `test/multiverse/suites/active_record/active_record_test.rb` tests MySQL in conjunction with ActiveRecord v3+ and performs the same assertion of expecting there to be 5 `SELECT` statements, so this appears to be a Postgres only issue.
* From examining the extra recorded `SELECT` statements that cause the test to fail, they all take place before the test's own database activity and they all appear to be related to Postgres' schema cache and TypeMap.
* The sometimes unexpected nature of these extra queries firing is an ongoing issue of discussion within the Rails community. See [Rails PR 44478](https://github.com/rails/rails/pull/44478) for example.

Now that we have identified the root cause of the test failures, we have two different routes available to tackle it.

* If we feel that the agent should not record these schema cache queries as counting towards the monitored app's query count, we can update the agent to stop doing so.
* If we wish to continue to count schema queries as monitored app queries, we'll have to refactor the assertion helper to support something like ">= 5" instead of simply "5" for the query count assertion, or do an even bigger testing overhaul to explicitly exclude queries that schema cache related.

I'll look at both approaches to give us some options.

**Comment by fallwith** [comment]
duplicate of #969
